apc_cache.c

```diff
@@ -1115,44 +1115,39 @@ PHP_APCU_API void apc_cache_stat(apc_cache_t *cache, zend_string *key, zval *sta
 /* {{{ apc_cache_defense */
 PHP_APCU_API zend_bool apc_cache_defense(apc_cache_t *cache, zend_string *key, time_t t)
 {
-	zend_bool result = 0;
-
 	/* only continue if slam defense is enabled */
 	if (cache->defend) {
 
 		/* for copy of locking key struct */
 		apc_cache_slam_key_t *last = &cache->header->lastkey;
 
-		if (!last->hash) {
-			return 0;
-		}
-
-		/* check the hash and length match */
-		if (last->hash == ZSTR_HASH(key) && last->len == ZSTR_LEN(key)) {
-			apc_cache_owner_t owner;
+		apc_cache_owner_t owner;
 #ifdef ZTS
-			owner = TSRMLS_CACHE;
+		owner = TSRMLS_CACHE;
 #else
-			owner = getpid();
+		owner = getpid();
```

Comment:
This will now always perform a getpid() call, even if hash/len do not match. It's probably not particularly expensive due to vDSO, but maybe best to avoid it?

Reply:
But we need to update the owner in the `last` entry or not?